### PR TITLE
Fix reclaim or preempt action may preempt driver of spark job

### PR DESCRIPTION
**Pull request by zbbkeepgoing**
fix: #2734

### CODE REVIEW

File: pkg/scheduler/plugins/gang/gang.go [hunk 1]
```diff
@@ -19,6 +19,8 @@ package gang
 import (
 	"fmt"
 
+	"volcano.sh/volcano/pkg/scheduler/plugins/util/gangextender"
```
Comment (wangyang0616) [inline pkg/scheduler/plugins/gang/gang.go:22]: It is suggested that the current imported information can be adjusted down a few lines and classified together

**Comment by wangyang0616** [comment]
I understand that it is not appropriate to put this in the gang scheduling strategy. Is it better to add an independent spark plugin?

**Comment by zbbkeepgoing** [comment]
> I understand that it is not appropriate to put this in the gang scheduling strategy. Is it better to add an independent spark plugin?

I don't think so, because for ordinary pgs, the min member of gang is the minimum number of all pods in pgs, but for applications like Spark, the driver is actually a part of the min member of gang

File: pkg/scheduler/plugins/gang/gang.go (hunk 1 above)
Comment (zbbkeepgoing) [reply pkg/scheduler/plugins/gang/gang.go]: Done

**Comment by wangyang0616** [comment]
reviewed 
cc @Yikun @william-wang

**Comment by hwdef** [comment]
still needs to be reviewed

**Comment by oemel09** [comment]
This should be configurable (e. g. via arguments to the gang plugin). There may be situations where preempting the spark driver is the desired behavior.

File: pkg/scheduler/plugins/util/gangextender/gangextender.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	sparkPodLabelRoleKey         = "spark-role"
+	sparkPodLabelRoleDriverValue = "driver"
+)
+
+func GangPreemptRejectExtender(preemptee *api.TaskInfo) bool {
+	sparkReject := sparkGangExtender(preemptee)
+	return sparkReject
+}
+
+func sparkGangExtender(preemptee *api.TaskInfo) bool {
```
Comment (lowang-bh) [inline pkg/scheduler/plugins/util/gangextender/gangextender.go:31]: I think the function name is not meaningful. May let the function name has something relative with "driver role" is easy to understand.

**Pull request closed without merging**